geode-core/src/main/java/org/apache/geode/internal/cache/execute/AllowExecutionInLowMemory.java

```diff
@@ -18,6 +18,6 @@
 /**
  * An internal marker interface used to allow functions to run in low-memory conditions.
  */
-public interface AllowExecutionInLowMemory extends InternalFunction<Object[]> {
+public interface AllowExecutionInLowMemory<T> extends InternalFunction<T> {
```

Comment:
Do we still need this interface? I'm a little confused about its purpose.

Reply:
It allows functions to run when Geode is under memory pressure. Have a look at `HeapMemoryMonitor.createLowMemoryIfNeeded`.

Reply:
Ah ok, I had forgotten about that.  Thank you and sorry for my confusion!